cylc/flow/data_messages.proto

```diff
@@ -117,7 +117,6 @@ message PbJob {
     optional string host = 15;
     optional string init_script = 16;
     optional string job_log_dir = 17;
-    optional string owner = 18;
```

Comment:
@dwsutherland good to remove like this?

Reply:
I think after the 8.0 final release it shouldn't happen. Instead we should probably use

```
optional string owner = 18 [deprecated=true];  // No longer used, see issue-1234; to be removed in 8.1 or 9.0...
```

But I think it should be OK to remove it now, but @dwsutherland is the best to answer that. Also, David, if this is removed now, do we re-organize the other field numbers (i.e. re-assign the `18` to the `19`, and so on?)?

Reply:
Yeah, that's fine.. Number tag doesn't need to be consecutive